java/code/src/com/suse/manager/webui/controllers/FormulaController.java

```diff
@@ -312,14 +312,14 @@ public static String saveSelectedFormulas(Request request, Response response,
                     FormulaFactory.saveGroupFormulas(id, selectedFormulas, user.getOrg());
                     break;
                 default:
-                    return errorResponse(response, Arrays.asList("Invalid target type!"));
+                    return errorResponse(response, Arrays.asList("error_invalid_target"));
             }
         }
         catch (IOException | UnsupportedOperationException e) {
             return errorResponse(response,
                     Arrays.asList("Error while saving formula data: " + e.getMessage()));
         }
-        return GSON.toJson(Arrays.asList("Formulas saved!"));
+        return GSON.toJson(Arrays.asList("formulas_saved"));
```

Comment:
I am not sure if we should do such a mapping of keys to messages here. As far as I understood we would simply use the english version of strings as keys for any future translation efforts. So is it maybe enough to move those strings to a constant in this class and treat `sid` or `sgid` as a parameter together with the respective ID?

Reply:
In this case the message is not a simple text but it contains also a link to the highstate page:
https://github.com/uyuni-project/uyuni/blob/c2a4ddddee9a95b669b8af132520819e8d505578/web/html/src/manager/group-formula-selection.js#L12
It looks more consistent to me to keep this in JS instead of hardcoding it in Java.

Reply:
It's okay for me, maybe we can get a confirmation from @ncounter?

Reply:
It is fine to me to have a mapping and a complex message. The only thing that is missing here is the translation `true-bypass` usage of the `t()` function. In the previous case we were returning simple text and could be translated at the very end of the code-path. Now we can't return the English version and translate it later because of the complex object returned.
This means instead that the visible text should go through [the t() function](https://github.com/uyuni-project/uyuni/blob/master/web/html/javascript/spacewalk-essentials.js#L436).
But because of the link object inside of the returned mapped value, I am not sure you can use the `argument replacement` implemented in the `t()` actually.

Reply:
Just to make sure, what I meant is: [this function](https://github.com/uyuni-project/uyuni/pull/897/files#diff-3482810fdbc75a64981d953b502ea8d8R16) will not do the expected job because the `messageTexts[msg]` will be a complex object instead of a simple text `String`. The translation needs to happen at the very begin of the object value creation then.

Reply:
@ncounter I've broken the string into several chunks and used `t(...)`.

Reply:
@mateiw great, I think that's fine for now as it is. Thanks :+1: